src/Operations.jl

```diff
@@ -428,7 +428,7 @@ function deps_graph(env::EnvCache, registries::Vector{Registry.RegistryInstance}
             # unregistered stdlib we must special-case it here.  This is further
             # complicated by the fact that we can ask this question relative to
             # a Julia version.
-            if is_unregistered_stdlib(uuid) || uuid_is_stdlib
+            if (is_unregistered_stdlib(uuid) || uuid_is_stdlib) && haskey(stdlibs(), uuid)
```

Comment:
I'm confused, as far as I can tell, this branch should already be `false`:

```
julia> using Pkg
       uuid = Base.UUID("a83860b7-747b-57cf-bf1f-3e79990d037f")
       stdlibs_for_julia_version = Pkg.Types.get_last_stdlibs(v"1.6.0")
       haskey(stdlibs_for_julia_version, uuid)
true

julia> stdlibs_for_julia_version = Pkg.Types.get_last_stdlibs(v"1.7.0")
       haskey(stdlibs_for_julia_version, uuid)
false

julia> Pkg.Types.is_unregistered_stdlib(uuid)
false
```

Reply:
But the error occurs when doing:

```julia
Pkg.add(Pkg.Types.Context(; julia_version = v"1.6"), [PackageSpec(; name = "libjulia_jll")])
```

So `uuid_is_stdlib` is true, and it tries to get the project file from an stdlib that is no longer there and crashes.